pypeit/core/wavecal/wvutils.py

```diff
@@ -199,81 +204,82 @@ def get_wave_grid(waves, masks=None, wave_method='linear', iref=0, wave_grid_min
     """
     c_kms = constants.c.to('km/s').value
 
-    if masks is None:
-        masks = waves > 1.0
-
-    if wave_grid_min is None:
-        wave_grid_min = waves[masks].min()
-    if wave_grid_max is None:
-        wave_grid_max = waves[masks].max()
-
-    dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
+    if wave_method == 'user_input':
+        wave_grid = wave_grid_input
+    else:
+        if masks is None:
+            masks = waves > 1.0
+
+        if wave_grid_min is None:
+            wave_grid_min = waves[masks].min()
+        if wave_grid_max is None:
+            wave_grid_max = waves[masks].max()
+
+        dwave_data, dloglam_data, resln_guess, pix_per_sigma = get_sampling(waves)
+
+        # TODO: These tests of the string value should not use 'in', they should use ==
```

Comment:
why?  `in` works the same and is nicer to read.

Reply:
TODO removed.

Reply:
I added that todo.  Previously (look in `develop`) this code has lines like, `elif 'linear' in wave_method`.  We should be testing equality in this case, not whether or not the string is a subset of the other string.  To my mind it's bad coding practice.  Here's a contrived, but somewhat relevant example:
```
>>> wave_method = 'log-linear'
>>> 'linear' in wave_method
True
>>> wave_method == 'linear'
False
```